docs/sql-keywords.md

```diff
@@ -117,6 +117,7 @@ Below is a list of all the keywords in Spark SQL.
   <tr><td>FALSE</td><td>reserved</td><td>non-reserved</td><td>reserved</td></tr>
   <tr><td>FETCH</td><td>reserved</td><td>non-reserved</td><td>reserved</td></tr>
   <tr><td>FIELDS</td><td>non-reserved</td><td>non-reserved</td><td>non-reserved</td></tr>
+  <tr><td>FILTER</td><td>non-reserved</td><td>non-reserved</td><td>reserved</td></tr>
```

Comment:
let's make it reserved under ansi mode. i.e. don't put the keyword in `ansiNonreserved`

Reply:
OK

Reply:
There exists a issue as below:
https://github.com/apache/spark/blob/bf7215c510e0eb2afa0a8f1337c535b4091f08f9/sql/core/src/test/resources/sql-tests/results/ansi/higher-order-functions.sql.out#L85
```
no viable alternative at input 'filter'(line 1, pos 7)
== SQL ==
select filter(ys, y -> y > 30) as v from nested
```
filter is a function.

Reply:
ah i see. We should put it in `functionName`, which means that it's a reserved keyword, but can be used as function name.

Reply:
OK. I learned it.